easypoll/src/main/java/com/eis0/easypoll/poll/BinaryPoll.java

```diff
@@ -0,0 +1,300 @@
+     */
+    public String getAuthorName() {
+        if(author.isLocalNetwork()) return SELF_AUTHOR_NAME;
+        return author.getAddresses().get(0);
+    }
+
+    /**
+     * Get the number of "Yes" for this poll.
+     *
+     * @return An integer representing the number of "Yes".
+     * @author Matteo Carnelos
+     */
+    public int getYesCount() {
+        return yesCount;
+    }
+
+    /**
+     * Get the number of "No" for this poll.
+     *
+     * @return An integer representing the number of "No".
+     * @author Matteo Carnelos
+     */
+    public int getNoCount() {
+        return noCount;
+    }
+
+    /**
+     * Tell if the poll is incoming, that means it is coming from another device.
+     *
+     * @return True if the poll is incoming, false otherwise.
+     * @author Matteo Carnelos
+     */
+    public boolean isIncoming() {
+        return incoming;
+    }
+
+    /**
+     * Tell if a poll is closed. A poll is closed when all the users have replied.
+     *
+     * @return True if the poll is closed, false otherwise.
+     * @author Matteo Carnelos
+     */
+    public boolean isClosed() {
+        return closed;
+    }
+
+    // ---------------------------- SETTERS ---------------------------- //
+
+    /**
+     * Set a user's answer and eventually close the poll.
+     *
+     * @param answer True if the answer is "Yes", false if the answer is "No".
+     * @author Matteo Carnelos
+     */
+    public void setAnswer(boolean answer) {
+        if(answer) yesCount++;
+        else noCount++;
+        closed = (yesCount + noCount == getUsersCount());
+    }
+
+    // ---------------------------- INSPECTIONS ---------------------------- //
+
+    /**
+     * Returns a percentage value representing the quantity of answers received in relationship
+     * with the total number of users that have to answer.
+     *
+     * @return The closed percentage as an integer value.
+     * @author Matteo Carnelos
+     */
+    public int getClosedPercentage() {
+        float answerCount = getYesCount() + getNoCount();
+        float ratio = answerCount / (float) getUsersCount();
+        return Math.round(ratio * 100);
+    }
+
+    /**
+     * Get the total number of users for this poll.
+     *
+     * @return The users count as an integer value.
+     */
+    public int getUsersCount() {
+        return users.size();
+    }
+
+    // ---------------------------- OVERRIDDEN METHODS ---------------------------- //
+
+    /**
+     * Compare two {@link BinaryPoll} objects and tell if they are equal. Two {@link BinaryPoll} are
+     * said to be equals if they have the same id.
+     *
+     * @param o The object to compare.
+     * @return True if the two objects are equal, false otherwise.
+     * @author Matteo Carnelos
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
I used an older version of your equals() method, just for curiosity why didn't you use the instanceof operator? Is it just because the result is the same, or you found this as a more proper way to do it?

Reply:
The `getClass()` way is more strict. With `instanceof` you will get `true` if one of the two classes is a subclass of the other, while with `getClass()` you will check that the two types are identical.
In this particular case using `instanceof` won't have caused any difference, but I usually prefer to be more strict on equals methods.

Reply:
I was hoping in this kind of answer. Thanks